### DISCUSSION TITLE
How to execute the RESTful command `get admin engine` as an admin user?

### DISCUSSION THREAD

**Discussion by HaoYang670** [apache/kyuubi#4671] [Q&A] [answered]
Hi Kyuubi community,

I met some problems when executing the `get admin engine` RESTful command, which needs your help.
I have created a Spark cluster with `share level = USER` and `user name = remziy`. But when sending the `get admin engine` REST command, I got the error: `Failed to validate proxy privilege of anonymous for remziy`.

My question is that how to let the sender be an admin who has privilege of all users, so that he/she can monitor the kyuubi engine states of all users as a real admin.

Thanks.

**Comment by yaooqinn** [comment]
cc @turboFei @zwangsheng

**Comment by zwangsheng** [comment]
First, Kyuubi Server provided `kyuubi.server.administrators` to manage administrator.

If you are requesting through the kyuubi rest client, then you can refer to the current authentication method. If Kyuubi Server uses the `NOSASL` authentication method, you can simply use the following code for as administrator.
```
KyuubiRestClient.builder("http://ip:port",)
  .authHeaderMethod(KyuubiRestClient.AuthHeaderMethod.BASIC)
  .username(ADMINISTRATORS_NAME)
  .build();
```

**Reply by HaoYang670** [reply]
Thank you, @zwangsheng, I will try your suggestion

**Reply by HaoYang670** [reply]
Hmm `kyuubi.server.administrator` is introduced in 1.8.0, isn't it? But I am using 1.7.0, unfortunately.

**Reply by zwangsheng** [reply, marked as answer]
Sorry for forget version affect, leaving aside the above administrator, you can still use
```
KyuubiRestClient.builder("http://ip:port",)
  .authHeaderMethod(KyuubiRestClient.AuthHeaderMethod.BASIC)
  .username("remziy")
  .build();
```

By the way, can you parse Kyuubi Server Log about why this error happened `Failed to validate proxy privilege of anonymous for remziy`

**Reply by HaoYang670** [reply]
Thank you @zwangsheng, it works
